domain/domain.go

```diff
@@ -3,15 +3,16 @@ package domain
 import (
 	"time"
 
+	"github.com/tiramiseb/go-gandi-livedns"
 	"github.com/tiramiseb/go-gandi-livedns/client"
 )
 
 type Domain struct {
 	client client.Gandi
 }
 
-func New(apikey string, sharing_id string, debug bool, dry_run bool) *Domain {
-	client := client.New(apikey,  sharing_id,  debug, dry_run)
+func New(apikey string, config *gandi.Config) *Domain {
```

Comment:
Pass gandi.Config (not a pointer to it).  New doesn't modify the values. https://github.com/golang/go/wiki/CodeReviewComments#pass-values "Don't pass pointers as function arguments just to save a few bytes. If a function refers to its argument x only as *x throughout, then the argument shouldn't be a pointer."